src/webrtc/call.ts

```diff
@@ -839,24 +867,30 @@ export class MatrixCall extends TypedEventEmitter<CallEvent, CallEventHandlerMap
                     // doesn't yet implement RTCRTPSender.setStreams()
                     // (https://bugzilla.mozilla.org/show_bug.cgi?id=1510802) so we'd have no way to group the
                     // two tracks together into a stream.
-                    const newSender = this.peerConn!.addTrack(track, callFeed.stream);
+                    const transceiver = this.peerConn!.addTransceiver(track, {
+                        streams: [callFeed.stream!],
+                        sendEncodings: isFirefox() ? undefined : encodings,
+                    });
 
-                    // now go & fish for the new transceiver
-                    const newTransciever = this.peerConn!.getTransceivers().find((t) => t.sender === newSender);
-                    if (newTransciever) {
-                        this.transceivers.set(tKey, newTransciever);
-                    } else {
-                        logger.warn("Didn't find a matching transceiver after adding track!");
+                    const parameters = transceiver.sender.getParameters();
+                    if (isFirefox()) {
```

Comment:
Could do with a comment on why the firefox-specific hackery is needed and what has to change so we can get rid of it, especially as it looks like we're doing the same unconditionally on line 856?

Reply:
Looking at all of this now, it seems we could probably remove some of the hackery but I am not fully confident yet - have created https://github.com/vector-im/element-call/issues/878 and will look at it ASAP

Reply:
Cool, that's good too.